### ISSUE TITLE
Nextseq reads

### ISSUE THREAD

**Issue by valery-shap** [nmquijada/tormes#34]
Hello again,

Before running TORMES I concat reads as:
zcat name_L001.fastq.gz .. name_L004.fastq.gz > all_R1.fastq
Then I started pipeline with these reads and it hangs on
Quality filtering process started at: 2020-12-06 22:44
	- Software for filtering chosen: prinseq
	- Minimum quality of reads to survive: 25
	- Minimum length of reads to survive: 125
There is no information about errors. How could it be so?

Best regards,
Valery

**Comment by nmquijada** [comment]
Hi Valery,

Could you please run your command again by redirecting the errors to a "error-tormes.txt" file and then share it with me?
For doing so:

```[YOUR TORMES COMMAND] &>>error-tormes.txt 2>>error-tormes.txt &```

Thanks!
Narciso

**Comment by valery-shap** [comment]
I run it on the server and I have 
#SBATCH --error=/home/user/test7.err
#SBATCH --output=/home/user/test7.log

[test7_err.txt](https://github.com/nmquijada/tormes/files/5651549/test7_err.txt)
[test7.log](https://github.com/nmquijada/tormes/files/5651550/test7.log)

**Comment by valery-shap** [comment]
[test8_err.txt](https://github.com/nmquijada/tormes/files/5651705/test8_err.txt)
[test8.log](https://github.com/nmquijada/tormes/files/5651707/test8.log)
This is without header error but still hanging.

**Comment by valery-shap** [comment]
This is for miseq reads that work good:
[test_err.txt](https://github.com/nmquijada/tormes/files/5651731/test_err.txt)

**Comment by valery-shap** [comment]
Also I've taken the 8 reads from server (4 for R1 and 4 for R2) to the local computer.
then I zcat them( zcat 1_file.fastq.gz 2_file.fastq.gz ... 4_file.fastq.gz > all.fastq.gz) , the output was .gz too. I had the error in TORMES with Trimmomatic with "Not in GZIP format"
I looked at these reads head -n 5 and the were really not gzipped
so I just renamed them to .fastq
and TORMES works!
I send these files to server and they work too.
Decided to repeat manipulations with zcat -> gz -> rename on the server and it hangs again on 
Quality filtering process started at: 2020-12-07 14:20
	- Software for filtering chosen: prinseq
	- Minimum quality of reads to survive: 25
	- Minimum length of reads to survive: 125
If I send the server zcat files to local computer, it hangs too...
sizes of files from local zcat and server zcat are the same .

**Comment by nmquijada** [comment]
Hi Valery,

Thanks for sharing the files.
Indeed, the problem seemt to be related to the compression of the files and with prinseq, as this software only accepts uncompressed fastq files to work. Trimmomatic and spades accept both compressed and uncompressed fastq files and therefore there was no error. 
That a file ends with ".gz" does not necessary mean that the file is gzipped. If you do ```mv file_L1.fastq file_L1.fastq.gz" you are changing the name of the file but it is not compressed.
You can inspect a file by using "file". If it is compressed it will return something like "gzip compressed data" and if not "ASCII text".
So, trimmomatic and spades worked because they wokr with both compress and uncompressed files (regarldess the name you gave to the file).

If you would like to concatenate gzipped files, you can use "cat": ```cat 1_file.fastq.gz 2_file.fastq.gz ... 4_file.fastq.gz > all.fastq.gz``` or "zcat" (plus further compression):
```
zcat 1_file.fastq.gz 2_file.fastq.gz ... 4_file.fastq.gz > all.fastq
gzip all.fastq
```

If you are concerned about the files ordering during the concatenation, you can do:
```
cat file_L1.fastq.gz > all.fastq.gz
cat file_L2.fastq.gz >> all.fastq.gz
cat file_L3.fastq.gz >> all.fastq.gz
...
```
The ">>" will add the the output of a command to the end of an already existing file.

If you do this in the server, are your nextseq files running properly?

Thanks,
Narciso

**Comment by valery-shap** [comment]
Hello, Narciso,

Thank you very much for replying so many times) Suppose I've collected all errors)
So my mistake was to concat nextseq reads NOT in the right order line1 -> line2 -> line3 -> line4.
Because of this TORMES was hanging on the step of the Prinseq. If TORMES could tell stupid users smth like "Oha la la your reads are wrong! Stop pipeline." , it'll be perfect.
and I used zcat -> .fastq Maybe it'll work with zcat -> .fastq.gz too. I'll test it in the future.
I've written a script that execute the zcat with the nextseq data folder and then it makes a metafile. 

Best regards,
Valery
